internal/impl/gcp/enterprise/input_spanner_cdc.go

```diff
@@ -0,0 +1,448 @@
+		return
+	}
+	if conf.StartTimestamp, err = parseRFC3339Nano(pConf, siFieldStartTimestamp); err != nil {
+		return
+	}
+	if conf.EndTimestamp, err = parseRFC3339Nano(pConf, siFieldEndTimestamp); err != nil {
+		return
+	}
+	if conf.HeartbeatInterval, err = parseDuration(pConf, siFieldHeartbeatInterval); err != nil {
+		return
+	}
+	if conf.MetadataTable, err = pConf.FieldString(siFieldMetadataTable); err != nil {
+		return
+	}
+	if conf.MetadataTable == "" {
+		conf.MetadataTable = fmt.Sprintf(defaultMetadataTableFormat, conf.StreamID)
+	}
+	if pConf.Contains(siFieldAllowedModTypes) {
+		if conf.AllowedModTypes, err = pConf.FieldStringList(siFieldAllowedModTypes); err != nil {
+			return
+		}
+	}
+	if conf.MinWatermarkCacheTTL, err = parseDuration(pConf, siFieldMinWatermarkCacheTTL); err != nil {
+		return
+	}
+
+	return
+}
+
+func spannerCDCInputSpec() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Beta().
+		Version("TODO").
+		Categories("Services", "GCP").
+		Summary("Creates an input that consumes from a spanner change stream.").
+		Description(`
+Consumes change records from a Google Cloud Spanner change stream. This input allows
+you to track and process database changes in real-time, making it useful for data
+replication, event-driven architectures, and maintaining derived data stores.
+
+The input reads from a specified change stream within a Spanner database and converts
+each change record into a message. The message payload contains the change records in
+JSON format, and metadata is added with details about the Spanner instance, database,
+and stream.
+
+Change streams provide a way to track mutations to your Spanner database tables. For
+more information about Spanner change streams, refer to the Google Cloud documentation:
+https://cloud.google.com/spanner/docs/change-streams
+`).
+		Field(service.NewStringField(siFieldCredentialsJSON).Optional().Description("Base64 encoded GCP service account JSON credentials file for authentication. If not provided, Application Default Credentials (ADC) will be used.").Default("")).
+		Field(service.NewStringField(siFieldProjectID).Description("GCP project ID containing the Spanner instance")).
+		Field(service.NewStringField(siFieldInstanceID).Description("Spanner instance ID")).
+		Field(service.NewStringField(siFieldDatabaseID).Description("Spanner database ID")).
+		Field(service.NewStringField(siFieldStreamID).Description("The name of the change stream to track")).
+		Field(service.NewStringField(siFieldStartTimestamp).Optional().Description("RFC3339 formatted timestamp to start reading from (default: current time)").Default("")).
+		Field(service.NewStringField(siFieldEndTimestamp).Optional().Description("RFC3339 formatted timestamp to stop reading at (default: no end time)").Default("")).
+		Field(service.NewStringField(siFieldHeartbeatInterval).Optional().Description("Duration string for heartbeat interval (e.g., '10s')").Default("10s")).
```

Comment:
Instead of saying `e.g. '10s'` there is a built in way to have examples per field using `.Example`

Reply:
Done.